src/sciline/visualize.py

```diff
@@ -1,53 +1,97 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
-from typing import Any, Callable, Dict, List, Tuple, get_args, get_origin
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    get_args,
+    get_origin,
+)
 
 from graphviz import Digraph
 
-from .scheduler import Graph
+from .pipeline import Pipeline, SeriesProvider
+from .typing import Graph, Item, Key
 
 
-def to_graphviz(graph: Graph, **kwargs: Any) -> Digraph:
+def to_graphviz(graph: Graph, compact: bool = False, **kwargs: Any) -> Digraph:
     """
     Convert output of :py:class:`sciline.Pipeline.get_graph` to a graphviz graph.
 
     Parameters
     ----------
     graph:
         Output of :py:class:`sciline.Pipeline.get_graph`.
+    compact:
+        If True, parameter-table-dependent branches are collapsed into a single copy
+        of the branch. Recommendend for large graphs with long parameter tables.
     kwargs:
         Keyword arguments passed to :py:class:`graphviz.Digraph`.
     """
     dot = Digraph(strict=True, **kwargs)
-    for p, (p_name, args, ret) in _format_graph(graph).items():
-        dot.node(ret, ret, shape='rectangle')
+    for p, (p_name, args, ret) in _format_graph(graph, compact=compact).items():
+        if '(' in ret and '=' not in ret:
```

Comment:
This seems brittle. Can you return an object with metadata from `_format_graph` instead of plain strings?